tests/runTests.ts

```diff
@@ -0,0 +1,129 @@
+import { RequestSpec, SpecResult, TestResult } from "../src/index";
+import { Tests } from "../src/index";
+
+function convertToString(item: any): string {
+  if (item === null) return "null";
+  if (item === undefined) return "undefined";
+  if (typeof item === "object") return JSON.stringify(item);
+  return item.toString();
+}
+
+function formatTestResult(res: TestResult, spec: string | null, skip?: boolean): string {
```

Comment:
Do we need formatting? I don't see the formatted result being printed anywhere on console. But maybe we should. As a module developer, I want to see what went wrong if anything did indeed go wrong by running this manually. Not just for CI, we would need something for manual execution before committing the code as well.

But the printing should be quite different from the CLI. Here we would print whether each test was expected to pass or fail, and whether it did pass or fail.

Reply:
If a test fails, we can click on it in GitHub and it shows us the console output. [Here's an example. ](https://github.com/agrostar/zzapi/actions/runs/9429421735/job/25975784030). 

If a test fails, I am displaying the test in a simple format (just name, spec, expected, received and message with a bullet point preceding it). You can try changing something in the test file, and running `npm run test` to see what the output would be.

Reply:
![Screenshot from 2024-06-22 15-00-24](https://github.com/agrostar/zzapi/assets/125901917/14e89f68-6832-4a7b-9682-7a65642afaf6)

This is an example. I changed tom to to in one test, and bar to ba.

Reply:
[Here](https://github.com/agrostar/zzapi/actions/runs/9624531895/job/26548370912) is how it looks in GitHub.